### PR TITLE
add a flag to indicate if a vindex handles null value

### PR DESCRIPTION
**Pull request by hihihuhu**
there is a recent change to improve `is null` query https://github.com/vitessio/vitess/pull/6133, it changed the behavior from always to fallback to scatter gather for null query, to treat null as an value and handle it in vindex. 
however, not all vindexes could handle null value, especially, the lookup table based vindex doesn't store null value at all,  so add a new flag to indicate if the vindex could handle null and fallback to scatter gather for null query if the vindex doesn't support it.

### CODE REVIEW

**Comment by harshit-gangal** [comment]
We already have function which does similar check directly use `NeedsVCursor` instead of `HasNullTest`

**Comment by hihihuhu** [comment]
thanks for pointing out! do you have a printer about the `NeedsVCursor` check? does it mean the issue of null query against lookup vindex always return empty result is fixed already

**Comment by harshit-gangal** [comment]
Inserted null value into lookup does not give any value. You can update your code to use NeedVCursor

**Comment by hihihuhu** [comment]
yes, this is not about inserting null value into lookup table, it is about query like `select * from table where col is null`, since lookup table doesn't have null value, the query needs to be a scatter gather, which is the behavior before the pr in the comment above.

**Comment by hihihuhu** [comment]
oh, i see your comment, yes, i could change to use `NeedVCursor`, however, just feel `NeedVCursor` may not necessary equal to cannot handle null value, though they could be the same thing for all exist vindexes

**Comment by hihihuhu** [comment]
updated to use `NeedVCursor`

**Comment by sougou** [comment]
I recently added support for "ignore_nulls" as an option for lookups here #6222. Will it be better if we just returned the full keyrange from the lookup vindex if this flag was set?

The other question is: did you want this for a unique or non-unique lookup?

One risk about a unique lookup is that there are places where we assume that no more than one shard will be a target. So, I'm concerned about how that interpretation may play out with a change like this.

Having said that, there is one rare (and intended) situation where a unique lookup does return more than one shard: https://github.com/vitessio/vitess/blob/master/go/vt/vtgate/vindexes/lookup.go#L227-L232. But the use case is acceptable because it happens while a lookup is being back-filled.

In other words, will it be ok if we changed this where a non-unique lookup returned a full scatter if input was `null` and `ignore_nulls` was set?
